normalizer/docker/parser.py

```diff
@@ -41,6 +43,42 @@ def __init__(
 
     def __str__(self):
         return self.content
+    
+    def get_build_args_envs(self, build_args_envs): 
+        build_args = '';
+        for index, item in enumerate(build_args_envs):
+            build_args += dedent(
+            f"""\
+            ARG {item}
+            """
+        )
+        build_envs = '';
+        for index, item in enumerate(build_args_envs):
+            build_envs += dedent(
+            f"""\
+            ENV {item}=${item}
+            """
+        )
+        return (build_args, build_envs)
+
+    def add_build_args_envs(self, build_args_envs):
+        build_args, build_envs = self.get_build_args_envs(build_args_envs)
+        if len(build_args):
+            self._parser.content+=build_args
+        if len(build_args):
+            self._parser.content+=build_envs
+    
+    def insert_build_args_envs(self, build_args_envs):
+        build_args, build_envs = self.get_build_args_envs(build_args_envs)
+        build_args_envs_str = build_args if len(build_args) else ''
+        build_args_envs_str += build_envs if len(build_args) else ''
+        if len(build_args_envs_str):
+            for env_var, var_name in FROM_VAR_RE.findall(self._parser.content):
+                self._parser.content = self._parser.content.replace(var_name, dedent(
+                f"""\
+                {var_name} 
+                """
+                )+build_args+build_envs)
```

Comment:
What's the purpose of this `FROM_VAR_RE`

Reply:
Find the FROM line in the dockerfile file, and write ARG and ENV into the dockerfile file